metabase/metabase
# Improve admin logs page loading and error experience

Fixes https://github.com/metabase/metabase/issues/24621

### Description

This PR introduces an error and no results states to the admin logs page. Previously the loading state would appear when one of these two scenarios occurred.

This PR also migrates the component to Typescript (I `git mv`ed the files but GitHub UI doesn't appear to be diffing it as such). I also remove the broken auto-scroll feature, fetch the logs 1 whole second earlier, and fix calling setState after unmount.

### How to verify

As mentioned in [the related issue](https://github.com/metabase/metabase/issues/24621), these states would only be a product of something being broken. You'll have to purposely break code to enter these states.

- go to admin settings
- go to troubleshooting (in navbar)
- go to logs (in sidebar)
- for verifying error state, replace the code block in `src/metabase/api/util.clj` on line 22 to return a 500 like so:
```
(api/defendpoint GET "/logs"
  "Logs."
  []
  (throw (ex-info "Some error message." {}))
  (validation/check-has-application-permission :monitoring)
  (logger/messages))
```

- for verifying no results state, it's easiest to either mock the endpoint or to hard code an empty `[]` result value in `Log.tsx`'s `fetchLogs` method

### Demo
Error view:
<img width="1491" alt="Screenshot 2024-01-10 at 1 12 14 PM" src="https://github.com/metabase/metabase/assets/7104357/70b21e70-4849-4a6d-9425-755a646a9619">

No results view:
<img width="1493" alt="Screenshot 2024-01-10 at 1 14 01 PM" src="https://github.com/metabase/metabase/assets/7104357/3e9f3f70-39b5-4a38-81f8-1f236a3afcf8">


### Checklist
